manifests/install.pp

```diff
@@ -62,6 +62,11 @@
           $_package_version = $package_version
         } else {
           $_package_version = "${package_version}-1${::lsbdistcodename}"
+          apt::pin { "pin ${::puppet_agent::package_name} package":
+            packages => $::puppet_agent::package_name,
+            priority => 1000,
```

Comment:
In order to support downgrade, my understanding is that priority should be 1001: https://linux.die.net/man/5/apt_preferences